cms/djangoapps/contentstore/views/course.py

```diff
@@ -137,7 +137,17 @@ def reindex_course_and_check_access(course_key, user):
     """
     if not has_course_author_access(user, course_key):
         raise PermissionDenied()
-    return CoursewareSearchIndexer.do_course_reindex(modulestore(), course_key)
+
+    course = modulestore().get_course(course_key)
```

Comment:
This feels like it should really be part of the indexing job, not part of the view. Also, with the changes for SOL-174, we are loading the full course as part of the indexing, so it will be much more efficient to do it therein.